Frameworks/UIKit/UINibUnarchiver.mm

```diff
@@ -203,7 +203,13 @@ static id constructObject(UINibUnarchiver* self, Object* pObj) {
         id classId = pObj->classType; // objc_getClass(pObj->className);
         assert(classId != nil);
 
-        pObj->cachedId = [classId alloc];
+        // TODO: #1105 alloc without init is invalid, and not all classes support alloc-init anyway
```

Comment:
As commented below, I can remove the one-off, but I'm still concerned about leaving [classId alloc] here:

- alloc without init is considered invalid
- we have a code path here (if classId doesn't support initWithCoder) where we effectively call [[classId alloc] autorelease].
- we appear to have been DEPENDING on this alloc autorelease code path, since if I change this to
    if ([classId instancesRespondToSelector:@selector(initWithCoder:)])
        pObj->cachedId = [[classId alloc] initWithCoder...]
  then things start exploding all over the place with an infinite recursion

we can take care of this separately in #1105, but I wanted to call it out while we're here and make sure you're aware of it

---
In reply to: [82918436](https://github.com/Microsoft/WinObjC/pull/1106#discussion_r82918436) [](ancestors = 82918436)